tests/Behat/PoetryTranslationContext.php

```diff
@@ -178,6 +180,78 @@ public function thePoetryTranslationRequestGetAccepted(string $title, string $la
     $this->performNotification($status_notification);
   }
 
+  /**
+   * Mock a Poetry notification with a status update for given content.
+   *
+   * The table has the form:
+   * | language  | status    |
+   * | Bulgarian | Ongoing   |
+   * | ...       | ...       |
+   *
+   * @param string $title
+   *   The node title.
+   * @param string $request_status
+   *   The demand status.
+   * @param \Behat\Gherkin\Node\TableNode $statuses
+   *   The status for each language.
+   *
+   * @When Poetry updates the status for :title as :request_status with the following individual statuses
+   */
+  public function poetrySendsStatusUpdate(string $title, string $request_status, TableNode $statuses): void {
+    $node = $this->getNodeByTitle($title);
+    $query = $this->getEntityJobsQuery($node);
+    $query->condition('job.state', Job::STATE_ACTIVE);
+    $query->condition('job.poetry_state', PoetryTranslator::POETRY_STATUS_ONGOING);
+    $result = $query->execute()->fetchAllAssoc('tjid');
+    $job = current(Job::loadMultiple(array_keys($result)));
+    if (empty($job)) {
+      throw new \Exception('A status update needs ongoing translations but none was found.');
+    }
+
+    $accepted = $refused = $cancelled = [];
+    $request_status_send = '';
+    foreach ($statuses->getHash() as $node_hash) {
+      $language = current($this->getLanguagesFromNames($node_hash['language']));
+      $status = $node_hash['status'];
+
+      $language_send = [
+        'code' => strtoupper($language->getId()),
+        'date' => '30/08/2050 23:59',
+        'accepted_date' => '30/09/2050 23:59',
+      ];
+
+      if ($request_status == 'Ongoing') {
+        $request_status_send = 'ONG';
```

Comment:
I think a bettter name would be $demand_status, since $request_status_send doesn't say much about that we are storing.

Reply:
I've choose "demand" word here before, then @upchuk have changed it. And I think it's ok like this.
We use request for this, demand seems more a french wording like we see in other places of the service. We're talking about the request status so it's an appropriate terminology.